app/assets/javascripts/discourse/app/components/user-menu/menu-tab.hbs

```diff
@@ -0,0 +1,31 @@
+{{#if this.renderAsLink}}
```

Comment:
Did we explore the possibility of just changing button action when active? and doing a `transitionTo("foo.bar")` in this case? With possible support for middle click if necessary. I generally prefer links if possible,  but here that seems quite hacky.

Reply:
Yeah I've thought about this and I don't think the switching elements approach is perfect, but I went with it because to me it felt more "right" from an accessibility perspective; I don't have anything to prove that this approach is better or worse. I couldn't find anything online about this so I don't really know ☹️

Reply:
Yeah, I think this is tricky either way because we're changing the behaviour of an element after it is first invoked. It could very well be that the "correct" approach is to avoid this scenario altogether (i.e. every click on an element has the same behaviour, always).

That said, flipping from `button` to `a` seems problematic for A11Y, we'd need to set focus to the replaced element. I suspect it might have UI repercussions too, small render jumps as elements are swapped plus long-term difficulties in maintaining styles for both types of elements.

I think we should try to keep the same element and keep the two behaviours (first click switches tabs, second click transitions to new route) and for screen readers we can maybe announce the change in a `role="status"` live region. I found this interesting example: https://adrianroselli.com/2021/01/multi-function-button.html, seems worth exploring @OsamaSayegh.

Reply:
I've changed this to keep the tab element unchanged when it's active (i.e. always a `<button>`), but I've not implemented the example because the new menu needs a bit of accessibility work for a later PR and I'll see how I can adapt the example to fit our use-case (I don't think it's quite what we're looking for, but pretty close).